dv/uvm/core_ibex/scripts/merge_cov.py

```diff
@@ -39,16 +39,22 @@ def find_cov_dbs(start_dir: pathlib.Path, simulator: str) -> Set[pathlib.Path]:
 
 
 def merge_cov_vcs(md: RegressionMetadata, cov_dirs: Set[pathlib.Path]) -> int:
-    logging.info("Generating merged coverage directory")
     cmd = (['urg', '-full64',
             '-format', 'both',
             '-dbname', str(md.dir_cov/'merged.vdb'),
             '-report', str(md.dir_cov/'report'),
             '-log', str(md.dir_cov/'merge.log'),
             '-dir'] +
            [str(cov_dir) for cov_dir in list(cov_dirs)])
-    return run_one(md.verbose, cmd, redirect_stdstreams='/dev/null')
 
+    with LockedMetadata(md.dir_metadata, __file__) as md:
+        md.cov_merge_log = md.dir_cov / 'merge.log'
+        md.cov_merge_stdout = md.dir_cov / 'merge.log.stdout'
+        md.cov_merge_cmds = [cmd]
+
+        with open(md.cov_merge_stdout, 'wb') as fd:
+            logging.info("Generating merged coverage directory")
+            return run_one(md.verbose, cmd, redirect_stdstreams=fd)
```

Comment:
No need to keep the Metadata file open while we actually perform the coverage merge.
```suggestion
    with LockedMetadata(md.dir_metadata, __file__) as md:
        md.cov_merge_log = md.dir_cov / 'merge.log'
        md.cov_merge_stdout = md.dir_cov / 'merge.log.stdout'
        md.cov_merge_cmds = [cmd]

    with open(md.cov_merge_stdout, 'wb') as fd:
        logging.info("Generating merged coverage directory")
        return run_one(md.verbose, cmd, redirect_stdstreams=fd)
```